### DISCUSSION TITLE
How to use emojis in nchat?

### DISCUSSION THREAD

**Discussion by 82MHz** [d99kris/nchat#112] [Q&A]
Hi d99kris,

EDIT: I'm leaving this here, but I am apparently blind, I just saw it's ctrl+c, written clearly on the screen :( 

~~when I open the emoji selector via ctrl+s, there seems to be no way to close it except selecting an emoji. Is this expected behaviour? Intuitively I would expect to be able to close it with either esc or ctrl+s again.~~

Related, most emojis for me show up as that rectangular unicode-symbol that shows that there is no symbol for it in my font set. How can I change that?

Thanks!

**Comment by d99kris** [comment]
Hi @82MHz - glad you found the command to cancel the emoji selection dialog. 👍 

> Related, most emojis for me show up as that rectangular unicode-symbol that shows that there is no symbol for it in my font set. How can I change that?

Do you know if your terminal / font supports those emojis? Let's say do they display properly if you google the emoji name (e.g. `1st_place_medal emoji`) and the copy the emoji 🥇 from the browser to the shell?

**Reply by 82MHz** [reply]
Thanks for your reply! I indeed didn't have the right emoji font installed. My terminal still refuses to display emojis, but this is a configuration problem on my end and nothing to do with nchat. So thanks for your help! :+1:

**Comment by Disonantemus** [comment]
> 1st_place_medal

Which font has all nchat emojis?

Right now I have the following installed fonts:
~~~
adobe-source-code-pro-fonts
cantarell-fonts
gnu-free-fonts
gsfonts
gsfonts-type1
libertinus-font
otf-ipafont
ttf-dejavu
ttf-jetbrains-mono-nerd
ttf-liberation
ttf-nerd-fonts-symbols
ttf-nerd-fonts-symbols-common
ttf-nerd-fonts-symbols-mono
~~~

Inside `nchat`, If I do `^s` to add emoji, just show a very few emojis and everything else is boxes (like I don't have any font with that unicode character)

![Looks like this to me](https://i.ibb.co/37dc2bD/20240311-114041-st.png)

Example: Paperclip (emoji)

[From this](https://unicodeplus.com/U+1F4CE) I got that "Paperclip" is: U+1F4CE
Checking for a matching font doesn't show anything:

    $ fc-list :charset=1f4ce

The weird thing is, if I do copy a character (like "paperclip") and paste into any "text input" (inside web browser, firefox), it get the icon/emoji/symbol correctly shown. And If I look in the [code](https://github.com/d99kris/nchat/blob/master/lib/ncutil/src/emojiutil_map.h), github shows all emojis.

**Reply by d99kris** [reply]
I'm not sure which fonts have better emoji coverage. Personally I'm using `Monospace` in `gnome-terminal` under `Ubuntu 22.04`, and `Monaco` in `Terminal` under `macOS`. Both have decent coverage I think.
